src/app/Card.tsx

```diff
@@ -1,33 +1,38 @@
 import { useEffect, useState } from 'react';
+import { FaTimes } from 'react-icons/fa';
+import { ansiFormat } from '../support/formatting';
+import { getWorkingDirectory } from '../support/nana';
 import { Output } from './Output';
 import { Prompt } from './Prompt';
-import { getWorkingDirectory } from '../support/nana';
-import { ansiFormat } from '../support/formatting';
-import { FaTimes } from 'react-icons/fa';
 
 export type CardPropTypes = {
   workingDir?: string;
-  input?: string;
+  input: string;
   output?: string;
 };
 
 export type ICard = CardPropTypes & {
-  id: number;
+  id: string;
 };
 
-export const Card = ({
-  input: initialInput,
-  workingDir,
-  history,
-  output,
-  onClose,
-  onSubmit,
-}: ICard & {
-  history: string[];
-  onSubmit: (newProps: Partial<CardPropTypes>, isError: boolean) => void;
-  onClose: () => void;
-}) => {
-  const [input, setInput] = useState(initialInput);
```

Comment:
I pulled this piece of state up. Now it just belongs to the card, and gets updated via `onInputChange` in App. The `activeHistoryIndex` state still belongs to this component, that hasn’t been pulled up yet. I’m not sure it needs to, though.

Reply:
Makes sense, thanks